### PR TITLE
feat: vault validator api

### PR DESCRIPTION
**Pull request by tyleroooo**

### CODE REVIEW

File: v4-client-js/src/clients/modules/composer.ts [hunk 1]
```diff
@@ -445,6 +450,51 @@ export class Composer {
     };
   }
 
+  // ------------ x/vault ------------
+  public composeMsgDepositToMegavault(
+    address: string,
+    subaccountNumber: number,
+    quoteQuantums: Uint8Array,
+  ): EncodeObject {
+    const subaccountId: SubaccountId = {
```
Comment (yogurtandjam) [inline v4-client-js/src/clients/modules/composer.ts:459]: [not related to this PR] wait, subaccountId is an object with an address and a number? feels like a weird name for this type.

File: v4-client-js/package.json
Comment (rosepuppy) [inline v4-client-js/package.json:1]: Need to version bump v4-client itself as well

Comment (tyleroooo) [reply v4-client-js/package.json:1]: just call the script? why?

Comment (tyleroooo) [reply v4-client-js/package.json:1]: wait isn't this automatically done now thanks to @yogurtandjam ?

Comment (rosepuppy) [reply v4-client-js/package.json:1]: oh I didn't know it was being done automatically

**Review by yogurtandjam — APPROVED** [review]

File: v4-client-js/package.json
Comment (yogurtandjam) [reply v4-client-js/package.json:1]: yeah @rosepuppy i think you were out but 
https://dydx-team.slack.com/archives/C03UV6K4KFY/p1726578615211089
I'll make sure to add some reference to this as a default PR description, or in the readme, or something

**Review by yogurtandjam — APPROVED** [review]

**Review by jaredvu — APPROVED** [review]

**Pull request merged**